### PR TITLE
Added deferred close to 'http.File' opened 

### LINKED ISSUE #9443: Report a coding issue that possibly leaks memory due to unclosed resources
**What did you see instead? Under which circumstances?**

At https://github.com/prometheus/prometheus/blob/main/web/web.go#L394, 
https://github.com/prometheus/prometheus/blob/1270b87970baeb926fcce64552db5c744ffaf83f/web/web.go#L394

File `f` is opened and then used to read the `idx`, however `f` (of type `http.File`) is not closed after the usage.
```
	serveReactApp := func(w http.ResponseWriter, r *http.Request) {
		f, err := ui.Assets.Open("/static/react/index.html")
		...
		idx, err := ioutil.ReadAll(f)
	}
```
Unlike `os.File` which has a finalizer associated with the file handle, `http.File` doesn't have such a finalizer: https://github.com/golang/go/blob/f0db7eae74ea235e9fbc2598252bfd46c1cc5510/src/net/http/fs.go#L754-L767

and it needs to be closed explicitly by invoking the closer, as demonstrated in: https://github.com/golang/go/blob/f0db7eae74ea235e9fbc2598252bfd46c1cc5510/src/net/http/fs.go#L597-L603

The unclosed file issue may cause memory issue since it is used multiple times within a loop:
https://github.com/prometheus/prometheus/blob/1270b87970baeb926fcce64552db5c744ffaf83f/web/web.go#L412-L413

**What did you expect to see?**

While this may not be a real issue depending on the underlying logic and the concrete resource and memory consumption, it is cleaner and safer to explicitly close the file after the usage:
```
	serveReactApp := func(w http.ResponseWriter, r *http.Request) {
		f, err := ui.Assets.Open("/static/react/index.html")
		defer f.Close()
		...
		idx, err := ioutil.ReadAll(f)
	}
```

**Found by static analyzer DeepGo**

### PR DESCRIPTION
**Pull request by kadern0**
Signed-off-by: Pablo Caderno <kaderno@gmail.com>

Fixes #9443

### CODE REVIEW

File: web/web.go [hunk 1]
```diff
@@ -392,6 +392,11 @@ func New(logger log.Logger, o *Options) *Handler {
 
 	serveReactApp := func(w http.ResponseWriter, r *http.Request) {
 		f, err := ui.Assets.Open("/static/react/index.html")
+		defer func() {
+			if err := f.Close(); err != nil {
+				fmt.Fprintf(w, "Error closing index.html: %v", err)
```
Comment (roidelapluie) [inline web/web.go:397]: At this stage, HTML has been served and HTTP code has been written. I would just ignore the error:

defer func(){ f.Close() }

Also, this should be called after we have done the err check below.

**Review by roidelapluie — COMMENTED** [review]
Thanks, added a few comments.

File: web/web.go (hunk 1 above)
Comment (kadern0) [reply web/web.go:397]: The linter complains if the error is unchecked. Do I ignore the error or the linter?

Comment (kadern0) [reply web/web.go:397]: Also, if ignoring the error, I could just use `defer f.Close()`,  right?

**Pull request merged**